src/Microsoft.AspNet.Mvc.Core/ModelBinding/CollectionModelBinder.cs

```diff
@@ -161,7 +188,18 @@ public class CollectionModelBinder<TElement> : IModelBinder
                 boundCollection.Add(ModelBindingHelper.CastOrDefault<TElement>(boundValue));
             }
 
-            return boundCollection;
+            return new CollectionResult
+            {
+                ValidationNode = validationNode,
+                Model = boundCollection
+            };
+        }
+
+        internal class CollectionResult
+        {
+            public ModelValidationNode ValidationNode { get; set; }
+
+            public IEnumerable<TElement> Model { get; set; }
```

Comment:
`ModelBindingResult` has more than what is required(viz. properties like `IsModelSet` and the `Key`), moreover the model is loosely typed in MBR, resulting in an unnecessary cast.